### PR TITLE
Updated armor cover calculation.

### LINKED ISSUE #2975: Wearing armors does not reduce damage when being attacked by mobs
Also the armor durability does not reduce after the attack

### PR DESCRIPTION
**Pull request by lkolbly**
Fixes #2975.

Updates how armor cover is calculated. Now has a function which calculates the armor cover and a function which calculates the armor cover from enchantments. Armor durability reduction is just from the amount blocked by the armor (not enchantments).

Has been tested with hits from other players (bare handed and with swords) and from zombies (bare handed), with the target player both wearing armor and not wearing armor.

### CODE REVIEW

File: Server/Plugins/APIDump/APIDesc.lua [hunk 1]
```diff
@@ -3019,6 +3019,17 @@ local Hash = cCryptoHash.sha1HexString("DataToHash")
...
+				ApplyArmorDamage =
+				{
+					Params =
+					{
+						{
+							Name = "DamageBlocked",
+							Type = "number",
+						},
+					},
+					Notes = "Applies durability to the armor, as if the armor blocked the given amount of damage.",
```
Comment (madmaxoft) [inline Server/Plugins/APIDump/APIDesc.lua]: `Lowers the armor's durability`...

File: src/Entities/Entity.h [hunk 2]
```diff
@@ -320,6 +320,9 @@ class cEntity
 	/** Returns the hitpoints out of a_RawDamage that the currently equipped armor would cover */
 	virtual int GetArmorCoverAgainst(const cEntity * a_Attacker, eDamageType a_DamageType, int a_RawDamage);
 
+	/** Returns the hitpoints that the current enchantments would cover */
```
Comment (madmaxoft) [inline src/Entities/Entity.h]: `currently equipped armor's enchantments`

Comment (lkolbly) [inline src/Entities/Entity.h]: Changed.

File: Server/Plugins/APIDump/APIDesc.lua (hunk 1 above)
Comment (lkolbly) [inline Server/Plugins/APIDump/APIDesc.lua]: Changed.

**Comment by tigerw** [comment]
CI crash fixed by #3866.

**Comment by tigerw** [comment]
Merge?

**Comment by lkolbly** [comment]
Just merged in the changes from #3866, confirmed this PR doesn't fail CI.

**Pull request merged**